fix: broken scripts

aidan46 opened the pull request:
This PR cleans up and modularizes the example bash scripts:

- Extracted shared logic (deal publishing, block fetching, piece CID/size calculation) into a new `deal_common.sh`.
- Extracted storage provider common logic into a new `sp_common.sh`.
- Removed redundant or hard-coded parameters.
- Scripts now fetch the latest finalized block height from the chain for start/end blocks.
- Made scripts quieter and easier to maintain.

Overall, this makes the RPC example flows cleaner, more maintainable, and easier to extend.

aidan46 committed:
fix: example scripts

jmg-duarte commented on examples/rpc_publish.sh:
```diff
@@ -28,92 +27,25 @@ INPUT_FILE="$1"
+    $START_BLOCK \
+    $END_BLOCK \
+    "5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY" \
+    "5FLSigC9HGRKVhB9FiEo4Y3koPsNmBmLJbpXg2mp1hXcS59Y" \
```
Who dis? Alice and Bob? Having their name here would be useful for whoever calls this to be able to check the chain

jmg-duarte commented on examples/rpc_publish_multiple_sectors.sh:
```diff
@@ -24,42 +24,24 @@ INPUT_FILE="$1"
+        $START_BLOCK \
+        $END_BLOCK \
+        "5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY" \
+        "5FLSigC9HGRKVhB9FiEo4Y3koPsNmBmLJbpXg2mp1hXcS59Y" \
```
Same

jmg-duarte reviewed (dismissed):
LGTM, comments I left are not crucial

aidan46 committed:
refactor: unify storage provider scripts

- Add sp_common.sh with shared helpers
- Refactor start_sp.sh and start_sps.sh to use common helpers
- Rename deal_functions.sh to deal_common.sh

aidan46 committed:
fix: increase end block for deal

aidan46 committed:
refactor: replace awk with jq for SS58 address extraction
